dbms/src/DataStreams/copyData.cpp

```diff
@@ -70,24 +70,41 @@ inline void doNothing(const Block &) {}
 
 void copyData(IBlockInputStream & from, IBlockOutputStream & to, std::atomic<bool> * is_cancelled)
 {
-    auto is_cancelled_pred = [is_cancelled] ()
-    {
+    auto is_cancelled_pred = [is_cancelled]() {
         return isAtomicSet(is_cancelled);
     };
 
     copyDataImpl(from, to, is_cancelled_pred, doNothing);
 }
 
+std::vector<Block> readAllData(IBlockInputStream & from, std::atomic<bool> * is_cancelled)
+{
+    from.readPrefix();
+    std::vector<Block> ret;
+
+    while (Block block = from.read())
+    {
+        if (isAtomicSet(is_cancelled))
+            break;
+        ret.push_back(block);
+    }
+
+    if (isAtomicSet(is_cancelled))
+        return {};
+
+    from.readSuffix();
```

Comment:
move to L91

Reply:
This piece of code is copied from `copyDataImpl`, in `copyDataImpl`, if the execution is cancelled, `from.readSuffix()` is not called.

Reply:
ok